library.c

```diff
@@ -2954,6 +2954,10 @@ redis_sock_check_liveness(RedisSock *redis_sock)
     smart_string cmd = {0};
     size_t len;
 
+    /* Sentinel doesn't have ECHO */
+    if (redis_sock->sentinel)
+        return SUCCESS;
+
```

Comment:
We have code for sentinel at the end of the function

Reply:
Found it. the logic is wrong in testing the errors. I will fix it correctly and push the change